pkg/ext/stores/tokens/tokens.go

```diff
@@ -327,39 +328,38 @@ func (t *Store) DeleteCollection(
 }
 
 // Delete implements [rest.GracefulDeleter], the interface to support the
-// `delete` verb. Delegates to the actual store method after some generic
-// boilerplate.
+// `delete` verb. Calls on `deleteCore`.
 func (t *Store) Delete(
 	ctx context.Context,
 	name string,
 	deleteValidation rest.ValidateObjectFunc,
 	options *metav1.DeleteOptions) (runtime.Object, bool, error) {
-	// locate resource first
-	obj, err := t.get(ctx, name, &metav1.GetOptions{})
+
+	userInfo, fullAccess, isRancherUser, err := t.auth.UserName(ctx, &t.SystemStore, "delete")
 	if err != nil {
-		return nil, false, err
+		return nil, false, err // The err is already an [apierrors.APIStatus].
 	}
 
-	// ensure that deletion is possible
-	if deleteValidation != nil {
-		err := deleteValidation(ctx, obj)
-		if err != nil {
-			return nil, false, err
-		}
+	// locate underlying resource first, bypass cache
+	secret, err := t.GetSecret(name, &metav1.GetOptions{}, false)
+	if err != nil {
+		return nil, false, err // The err is already an [apierrors.APIStatus].
 	}
 
-	// and now actually delete
-	if err := t.delete(ctx, obj, options); err != nil {
-		return nil, false, err
+	if !fullAccess && (!isRancherUser || !userMatchSecret(userInfo.GetName(), secret)) {
+		// An ordinary user can only access their own tokens.  We return
+		// a NotFound error to avoid leaking information about other
+		// users' tokens.
+		return nil, false, apierrors.NewNotFound(GVR.GroupResource(), name)
 	}
 
-	return obj, true, nil
+	return t.deleteCore(ctx, secret, deleteValidation, options)
 }
 
-// deleteToken is a variant of Delete for use by DeleteCollection. It avoids
-// doing a second `token get` calling on `Delete` would invoke. Delegates to the
-// actual store method after some generic boilerplate.
-func (t *Store) deleteToken(
+// deleteCore is invoked by Delete and DeleteCollection, after access checks,
+// with the backing secret of the token to delete. After conversion and further
+// checks it delegates to the system store.
```

Comment:
Ideally we want to avoid comments like this as it presents self evident information like what calls it or what it delegates to, which also can and will easily go out-of-date, but never says what it does or why. Maybe just remove it?